packages/router-core/src/new-process-route-tree.ts

```diff
@@ -0,0 +1,638 @@
+					const suffix_raw = path.substring(data[4]!, end)
+					const prefix = !prefix_raw ? undefined : caseSensitive ? prefix_raw : prefix_raw.toLowerCase()
+					const suffix = !suffix_raw ? undefined : caseSensitive ? suffix_raw : suffix_raw.toLowerCase()
+					const existingNode = node.dynamic?.find(s => s.caseSensitive === caseSensitive && s.prefix === prefix && s.suffix === suffix)
+					if (existingNode) {
+						nextNode = existingNode
+					} else {
+						const next = createDynamicNode(SEGMENT_TYPE_PARAM, route.fullPath, caseSensitive, prefix, suffix)
+						nextNode = next
+						next.depth = ++depth
+						next.parent = node
+						node.dynamic ??= []
+						node.dynamic.push(next)
+					}
+					break
+				}
+				case SEGMENT_TYPE_OPTIONAL_PARAM: {
+					const prefix_raw = path.substring(start, data[1])
+					const suffix_raw = path.substring(data[4]!, end)
+					const prefix = !prefix_raw ? undefined : caseSensitive ? prefix_raw : prefix_raw.toLowerCase()
+					const suffix = !suffix_raw ? undefined : caseSensitive ? suffix_raw : suffix_raw.toLowerCase()
+					const existingNode = node.optional?.find(s => s.caseSensitive === caseSensitive && s.prefix === prefix && s.suffix === suffix)
+					if (existingNode) {
+						nextNode = existingNode
+					} else {
+						const next = createDynamicNode(SEGMENT_TYPE_OPTIONAL_PARAM, route.fullPath, caseSensitive, prefix, suffix)
+						nextNode = next
+						next.parent = node
+						next.depth = ++depth
+						node.optional ??= []
+						node.optional.push(next)
+					}
+					break
+				}
+				case SEGMENT_TYPE_WILDCARD: {
+					const prefix_raw = path.substring(start, data[1])
+					const suffix_raw = path.substring(data[4]!, end)
+					const prefix = !prefix_raw ? undefined : caseSensitive ? prefix_raw : prefix_raw.toLowerCase()
+					const suffix = !suffix_raw ? undefined : caseSensitive ? suffix_raw : suffix_raw.toLowerCase()
+					const next = createDynamicNode(SEGMENT_TYPE_WILDCARD, route.fullPath, caseSensitive, prefix, suffix)
+					nextNode = next
+					next.parent = node
+					next.depth = ++depth
+					node.wildcard ??= []
+					node.wildcard.push(next)
+				}
+			}
+			node = nextNode
+		}
+		if (route.path)
+			node.routeId = route.id
+		onRoute(route, node)
+	}
+	if (route.children) for (const child of route.children) {
+		parseSegments(data, child as TRouteLike, cursor, node, depth, onRoute)
+	}
+}
+
+function sortDynamic(a: { prefix?: string, suffix?: string }, b: { prefix?: string, suffix?: string }) {
+	if (a.prefix && b.prefix && a.prefix !== b.prefix) {
+		if (a.prefix.startsWith(b.prefix)) return -1
+		if (b.prefix.startsWith(a.prefix)) return 1
+	}
+	if (a.suffix && b.suffix && a.suffix !== b.suffix) {
+		if (a.suffix.endsWith(b.suffix)) return -1
+		if (b.suffix.endsWith(a.suffix)) return 1
+	}
+	if (a.prefix && !b.prefix) return -1
+	if (!a.prefix && b.prefix) return 1
+	if (a.suffix && !b.suffix) return -1
+	if (!a.suffix && b.suffix) return 1
+	return 0
+}
```

Comment:
should rank case-sensitive routes w/ prefix/suffix higher than case-insensitive ones